### ISSUE TITLE
Have anyone meet the issue when onSwipedAll update new cards list but component not re-render?

### ISSUE THREAD

**Issue by iita71737** [webraptor/react-native-deck-swiper#72]
`const UserMatchCardSwiper = (props) => {
  const {width, height} = Dimensions.get('window');
  const {t, i18n} = useTranslation();

  const {
    navigation,
    cards,
    onSwipedAll
  } = props

  // dummy data
  const [dummyCards, setDummyCards] = React.useState([
    {
      firstName: 'Wasa ',
      lastName: 'John',
      job: 'Software Developer',
      photoURL:
        'https://the-allstars.com/blog/images/articl/1513149176673779.jpg',
      age: 27,
      id: 1,
    },
    {
      firstName: 'Alan ',
      lastName: 'Chen',
      job: 'Attack on titan',
      photoURL:
        'https://www.mirrormedia.com.tw/assets/images/20210811183042-492063c52c4c70e0ffe94db30f8395b8-mobile.jpg',
      age: 29,
      id: 2,
    },
    {
      firstName: 'David ',
      lastName: 'Turtle',
      job: 'SuperMan',
      photoURL:
        'https://thumbor.4gamers.com.tw/jAvXlYYj5FJ85bbk-3XR77UORfk=/800x0/filters:extract_cover():no_upscale():quality(80):format(jpeg):background_color(FFFFFF)/https%3A%2F%2Fimg.4gamers.com.tw%2Fpuku-clone-version%2F362610ba3e486edce20dc30abfa265e270af5d50.png',
      age: 30,
      id: 3,
    }
  ])

  // States
  const [page, setPage]  = React.useState(1)

  // Services
  const $_SwipeLeft = async(cardIndex) => {
    // const _cardId = cards[cardIndex].id
    // const res = await S_User.I_UnLike(_cardId)
  }

  const $_SwipeRight = async(cardIndex) => {
    const _cardId = cards[cardIndex].id
    const res = await S_User.I_Like(_cardId)
  }

  React.useEffect(()=>{
    if(page){
      onSwipedAll(page)
    }
  },[page])

  return (
    <View
      style={{
        flex: 1,
        marginTop: -48,
      }}>
      <Swiper
        useViewOverflow={Platform.OS === 'ios'}
        containerStyle={{backgroundColor: 'transparent'}}
        cards={dummyCards}
        stackSize={3}
        cardIndex={0}
        animateCardOpacity
        verticalSwipe={false}
        onSwipedLeft={(cardIndex) => {
          $_SwipeLeft(cardIndex)
        }}
        onSwipedRight={(cardIndex) => {
          $_SwipeRight(cardIndex)
        }}
        onSwipedAll={() => {
          console.log('onSwipedAll')
          const _test = [
            {
            firstName: 'QQQQQ123 ',
            lastName: 'John',
            job: 'Software Developer',
            photoURL:
              'https://agirls.aottercdn.com/media/87ddac33-a43a-4c34-86a4-c2a508ef68a7.jpg',
            age: 27,
            id: 98,
          },
          {
            firstName: 'QQQQ321 ',
            lastName: 'Chen',
            job: 'Attack on titan',
            photoURL:
              'https://www.mirrormedia.com.tw/assets/images/20210811183042-492063c52c4c70e0ffe94db30f8395b8-mobile.jpg',
            age: 29,
            id: 97,
          }]
          setDummyCards(_test)
          // setPage(page + 1)
        }}
        overlayLabels={{
          left: {
            title: 'NOPE',
            style: {
              label: {
                textAlign: 'right',
                color: 'red',
              },
            },
          },
          right: {
            title: 'LIKE',
            style: {
              label: {
                textAlign: 'left',
                color: 'green',
              },
            },
          },
        }}
        infinite={false}
        keyExtractor={(card, index)=> index}
        renderCard={(card, index)=> (
          <>
            <View><WsText>{card.firstName}</WsText></View>
            {/* <UserMatchCard001 key={index} card={card} navigation={navigation}></UserMatchCard001> */}
          </>
        )}
      />
    </View>
  );
};
export default UserMatchCardSwiper;
`

**Comment by Mukthayar1** [comment]
yes having same issues

**Comment by solonik1** [comment]
Yes, same for me

**Comment by daviddalmaso** [comment]
Same for me

**Comment by RuzenDev** [comment]
Yes, After finisihing the swipe. I am needing to get new card arrays data but its not working. 

has anybody solved this????? @webraptor

**Comment by Riaz5687** [comment]
same for me

**Comment by perroudsky** [comment]
same
